src/fortranscanner.l

```diff
@@ -85,6 +86,13 @@ typedef yyguts_t *yyscan_t;
 enum ScanVar { V_IGNORE, V_VARIABLE, V_PARAMETER, V_RESULT};
 enum InterfaceType { IF_NONE, IF_SPECIFIC, IF_GENERIC, IF_ABSTRACT };
 
+struct blockState
+{
+  blockState(QCString str, int lineNr) : blockString(str), blockLineNr(lineNr) {}
+  QCString blockString;
+  int      blockLineNr = -1;
+};
+std::stack<blockState> g_blockStack;
```

Comment:
Global variables will create data races and crashes when using multiple threads (e.g. `NUM_PROC_THREADS!=1`). 
Assuming this state is per file, you need to put this variable in `fortranscannerYY_state` and access it via `yyextra`.

Reply:
I think the commit e87e708 from February 17, 2024 fixes this problem.